### PR TITLE
Websocket/order update/sss 75

### PR DESCRIPTION
**Pull request by ACC-Vaibhav-Palvi**
New URI update for order update websocket.

### CODE REVIEW

File: src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+
+@Slf4j
+public class OrderUpdateWebsocket {
+    private static final int PING_INTERVAL = 10000; // 10 seconds
```
Comment (anurag-amx) [inline src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java]: append ms in variable name

File: src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    private static final String HEADER = "Authorization";
```
Comment (anurag-amx) [inline src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java]: header name should be HEADER_AUTHORIZATION

File: src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    private final String wsuri = routes.getOrderUpdateUri();
+    private WebSocket ws;
+    private String accessToken;
+    private final OrderUpdateListner orderUpdateListner;
+
+    private Timer pingTimer;
+    private LocalDateTime lastPongReceivedTime = LocalDateTime.now();
+
+    public OrderUpdateWebsocket(String accessToken, OrderUpdateListner orderUpdateListner) {
+        if (accessToken.isEmpty() || Utils.validateInputNullCheck(orderUpdateListner)) {
```
Comment (anurag-amx) [inline src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java]: use Strings.empty

File: src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                    LocalDateTime currentTime = LocalDateTime.now();
+                    if (lastPongReceivedTime.isBefore(currentTime.minusSeconds(20))) {
+                        websocket.disconnect();
+                        reconnect();
+                    }
+                } catch (Exception e) {
+                    orderUpdateListner.onError(getErrorHolder(e));
+                }
+            }
+        }, 5000, 5000); // run at every 5 second
```
Comment (anurag-amx) [inline src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java]: dont use constants value here.

**Review by anurag-amx — CHANGES_REQUESTED** [review]

File: src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java (hunk 1 above)
Comment (ACC-Vaibhav-Palvi) [reply src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java]: Variables names are changed accordingly. Please check

File: src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java (hunk 2 above)
Comment (ACC-Vaibhav-Palvi) [reply src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java]: Header variable is changed accordingly. Please check

File: src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java (hunk 3 above)
Comment (ACC-Vaibhav-Palvi) [reply src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java]: now using StringUtils.isEmpty() which will check both empty and null condition.

File: src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java (hunk 4 above)
Comment (ACC-Vaibhav-Palvi) [reply src/main/java/com/angelbroking/smartapi/orderupdate/OrderUpdateWebsocket.java]: constant values are now place inside a global variable.

**Review by anurag-amx — APPROVED** [review]

**Review by anurag-amx — DISMISSED** [review]

**Review by gulshan178 — APPROVED** [review]

**Pull request merged**